src/libopensc/reader-pcsc.c

```diff
@@ -2032,7 +2031,7 @@ static int part10_build_modify_pin_block(struct sc_reader *reader, u8 * buf, siz
 	if (!pin_ref->min_length || !pin_ref->max_length)
 		return SC_ERROR_INVALID_ARGUMENTS;
 
-	tmp16 = (pin_ref->min_length << 8 ) + pin_ref->max_length;
+	tmp16 = (unsigned)((pin_ref->min_length << 8 ) + pin_ref->max_length);
```

Comment:
```suggestion
	tmp16 = (uint16_t)(((pin_ref->min_length << 8 )  & 0xFF00) + (pin_ref->max_length) & 0xFF);
```

Reply:
:facepalm: indeed. Thanks for careful review and patience!